Sources/Aardvark/BugReporting/DictionaryAttachmentGenerator.swift

```diff
@@ -16,25 +16,33 @@
 
 import Foundation
 
-@objc(ARKUserDefaultsAttachmentGenerator)
-public final class UserDefaultsAttachmentGenerator: NSObject {
+@objc(ARKDictionaryrAttachmentGenerator)
+public final class DictionaryAttachmentGenerator: NSObject {
 
-    /// Generates an attachment containing a representation of the data in the specified user defaults store.
+    /// Generates a plist attachment containing key-value pairs.
     ///
-    /// - parameter userDefaults: The user defaults store from which to pull data.
+    /// - parameter keyValueStore: A dictionary representing the key-value store from which the attachment will be created.
+    /// - parameter keys: An optional list of keys. If provided, only the key-value pairs with keys in this list will be included in the attachment. If not, all key-value pairs from the `keyValueStore` will be included.
     /// - parameter fileName: An optional file name for the attachment, without the file extension.
     public static func attachment(
-        for userDefaults: UserDefaults = .standard,
-        named fileName: String? = nil
+        for keyValueStore: [String: Any],
+        filter keys: [String]? = nil,
```

Comment:
nit: Let's rename this label to `includedKeys` instead of `filter`. I think filter is a bit ambiguous as to whether you're filtering down to these keys or filtering out these keys.

Reply:
Goodness, I scratched my head for so long to find a better name and it was right there 😅